### PR TITLE
Catch incorrect tax number error

### PR DESCRIPTION
**Pull request by frederike-ramin**
# Short Description
- We want to explicitly catch the only known error that we have left coming directly from Elster, which results in an unpretty error message.
- This also introduces a new design for the filing failure page.

# Changes
- Create InvalidTaxNumberError as a specific case of ValidationError
- Raise this in erica, instead of the more general ValidationError
- Catch that in the webapp and set the correct error message
- Move the failure template with icon into the base templates, make the eligibility failure template a child of that 
- Separated the eligibility.css into two files, to separate the css for failure/success page from the more general classes

# Feedback
- Maybe spend a minute to think about whether this potentially breaks (the display) of other error messages that we might not yet catch.
- Does the setting of the error message within the error make sense to you?
- What do you think of the way the `title` and `error_details` are set in the templates? Would you prefer them to be set in the step?

### CODE REVIEW

File: webapp/app/templates/lotse/display_filing_failure.html [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
-{% block display_content %}
-    <div class="alert alert-danger" role="alert">
-        <h1 class="alert-heading">{{ _('form.lotse.filing.failure.alert.title') }}</h1>
-        {% for error in elster_data['validation_problems'] %}
-            <ul>
-                <li>{{ error }}</li>
-            </ul>
-        {% endfor %}
+{% set title = _('form.lotse.filing.failure.alert.title') %}
+{% set error_details = elster_data['validation_problems'] %}
```
Comment (JulianRoesner) [inline webapp/app/templates/lotse/display_filing_failure.html:4]: Okay, to be honest this confused me a little. Because this is not set in the `display_failure_icon` I was a little confused. I then overlooked that it is set in `display_failure`. 

Maybe we either should name it a little bit more explicit here (`validation_problem`?) or we do set it in the step.

**Review by JulianRoesner — APPROVED** [review]
I assumed that you would solve it quite differently (adding a new error handler and a specific error page for that kind of error). However, I think it makes a lot more sense to catch the error. Nevertheless, I am wondering if we might find a way to make it more explicit.
So, take my remarks more as a styling discussion.

> Maybe spend a minute to think about whether this potentially breaks (the display) of other error messages that we might not yet catch.
I do not think that there is a direct problem with that approach. I have also tested all the different errors coming from Erica and everything looked fine.
That said, I am not so sure if the new `display_failure`pages might not be too clever. I know that I have said otherwise at some other point in the past, but I think here all of the inheritance in the Jinja templates and passing all the information from the error the the pages makes it quite hard to follow along.
We could leave it like that or we might make the error pages more explicit. Although they share a lot of information.

Besides that the `erica_error.html` does look like before. I think that is no big problem because it only occurs for all unknown errors.

> Does the setting of the error message within the error make sense to you?
I am not so sure about it. It makes sense. However, it changes a little how we use the `ElsterGlobalValidationError`. The attribute `validation_problems` was so far used to set validation problems coming from Erica.
Maybe we could just use a separate class here.

> What do you think of the way the title and error_details are set in the templates? Would you prefer them to be set in the step?
To be honest, it took me some time to figure out what is happening (s. comment).

File: webapp/app/templates/eligibility/display_success.html [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 {% block additional_stylesheets %}
     {{ super() }}
+    <link rel="stylesheet" href="{{ static_url_for('static', filename='css/display_failure_success.css') }}">
```
[the remark this answers is not in the document]
Comment (frederike-ramin) [reply webapp/app/templates/eligibility/display_success.html:7]: I'm not sure I get what you mean. I just have one stylesheet for failure and success pages?

File: webapp/app/translations/de/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -125,10 +125,21 @@ msgid "500.header-title"
 msgstr "Fehler 500 - Der Steuerlotse für Rente und Pension"
 
 #: app/elster_client/elster_errors.py:99
-msgid "form.lotse.input_invalid.InvalidTaxNumber"
+#, fuzzy
```
[the remark this answers is not in the document]
Comment (frederike-ramin) [reply webapp/app/translations/de/LC_MESSAGES/messages.po:128]: Definitely did.

File: webapp/app/elster_client/elster_client.py [hunk 4]
```diff
@@ -237,6 +236,8 @@ def check_pyeric_response_for_errors(pyeric_response):
                                               server_response=server_response)
         elif error_code == 12:
             raise ElsterInvalidBufaNumberError()
+        elif error_code == 13:
+            raise ElsterInvalidTaxNumberError(message=error_message, eric_response=eric_response)
```
[the remark this answers is not in the document]
Comment (frederike-ramin) [reply webapp/app/elster_client/elster_client.py:240]: It inherits from ValidationError, so yes, I never catch it directly. I actually thought it's quite smooth. :D

File: webapp/app/templates/lotse/display_filing_failure.html (hunk 1 above)
Comment (frederike-ramin) [reply webapp/app/templates/lotse/display_filing_failure.html:4]: I we'd "name it a little bit more explicit here (validation_problem?)" here, we'd need to also rename it in `eligibility/display_failure.hmtl` and I don't think that's a good idea, that's why I chose the more general name of "error_details". But yes, we can set it in the step's `render` method if you prefer that. I'll then also set it in the eligibility failure steps, instead of in `eligibility/display_failure.hmtl`.

File: webapp/app/templates/eligibility/display_success.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (JulianRoesner) [reply webapp/app/templates/eligibility/display_success.html:7]: I was thinking about two different stylesheets. But mainly the problem I had was that the naming implied to me the `display_failure` is the general version and `display_failure_icon` the special case.

File: webapp/app/templates/lotse/display_filing_failure.html (hunk 1 above)
Comment (JulianRoesner) [reply webapp/app/templates/lotse/display_filing_failure.html:4]: Let's maybe quickly talk about it

File: webapp/app/templates/eligibility/display_success.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (frederike-ramin) [reply webapp/app/templates/eligibility/display_success.html:7]: Okay, I renamed the file to `eligibility/display_failure_icon.html`.

File: webapp/app/templates/lotse/display_filing_failure.html (hunk 1 above)
Comment (JulianRoesner) [reply webapp/app/templates/lotse/display_filing_failure.html:4]: We've decided to rename it to `eligibility/display_failure_with_icon.html`
